src/main/java/com/cloudbees/plugins/credentials/CredentialsStoreAction.java

```diff
@@ -196,8 +197,9 @@ public HttpResponse doCreateDomain(StaplerRequest req) throws ServletException,
         getStore().checkPermission(MANAGE_DOMAINS);
         JSONObject data = req.getSubmittedForm();
         Domain domain = req.bindJSON(Domain.class, data);
-        if (getStore().addDomain(domain)) {
-            return HttpResponses.redirectTo("./domain/" + Util.rawEncode(domain.getName()));
+        String domainName = domain.getName();
+        if (domainName != null && getStore().addDomain(domain)) {
```

Comment:
@halkeye filed #47 to address the front-end part too. Let's see what the community says.